src/Voron/Impl/EncryptionBuffersPool.cs

```diff
@@ -77,7 +79,8 @@ public void Return(byte* ptr, long size, NativeMemory.ThreadStats allocatingThre
             size = Bits.PowerOf2(size);
             Sodium.sodium_memzero(ptr, (UIntPtr)size);
 
-            if (Disabled || size > _maxBufferSizeToKeepInBytes || (_isLowMemory.IsRaised() && generation < Generation))
+            if (Disabled || size / Constants.Storage.PageSize > MaxNumberOfPagesToCache || 
```

Comment:
You are modified the `size` and If the size is larger than 1MB, Will you return here more than you allocated??

Reply:
And zero it beyond the range.

Reply:
It is fine to return more than the requested size, yes.
We aren't zeroing the data here, the caller will do that when they return the data back.